controller/common/lut.py

```diff
@@ -0,0 +1,78 @@
+from typing import List, Union
+
+import numpy as np
+from scipy import interpolate
+
+
+class LUT:
+    """
+    Class for performing look-up table interpolation.
+
+    Methods:
+        __init__: Initializes the LUT object with lookup table data and interpolation method.
+        __call__: Calls the interpolation method with the given input.
+    """
+
+    def __init__(
+        self,
+        lookup_table: Union[List[List[int | float]], np.ndarray],
+        interpolation_method: str = "linear",
+    ):
+        """
+        Initializes the LUT object.
+
+        Args:
+            lookup_table (list): A list of tuples or lists containing x-y data points for
+            interpolation. Format is [[x1,y1], [x2,y2], ..., [xn,yn]]
+            interpolation_method (str): Interpolation method to use. Default is "linear".
+
+        Raises:
+            ValueError: If the specified interpolation method is unknown.
+        """
+        if type(lookup_table) is np.ndarray:
+            self.__table = lookup_table
+        else:
+            self.__table = np.array(lookup_table)
+
+        self.__verifyTable(self.__table)
+        self.x = self.__table[:, 0]
+        self.y = self.__table[:, 1]
+
+        self.__interpolation = interpolation_method
+
+        match self.__interpolation:
+            case "linear":
+                self.__method = self.__linearInterpolation
```

Comment:
I would rename this variable to something more descriptive. Something like `self.__interpolation_function`

Reply:
renamed them:
```

        match self.__interpolation_method:
            case "linear":
                self.__interpolation_function = self.__linearInterpolation
            case "spline":
                self.__interpolation_function = self.__splineInterpolation
            case _:
                raise ValueError(
                    self.__interpolation_method + " is an unknown interpolation method!"
                )

```